### ISSUE TITLE
Preload script loaded but not working in production

### ISSUE THREAD

**Issue by oz-** [nklayman/vue-cli-plugin-electron-builder#1194]
I have this preload script which is **working like a charm in development mode**, and while **it is loaded in production mode**, it outputs an error: **'require is not defined'**

I guess this is more a webpack configuration problem, but I can't understand what.

### The preload.js script
```
// With contextIsolation: true
import { contextBridge, ipcRenderer } from 'electron'
// const { contextBridge, ipcRenderer } = require('electron')

contextBridge.exposeInMainWorld('electron', {
    send: (channel, ...args) => {
      ipcRenderer.send(channel, ...args)
    },
    on: (channel, listener) => {
      ipcRenderer.on(channel, listener)
    },
    off: (channel, listener) => {
      ipcRenderer.removeEventListener(channel, listener)
    }
  }
)
```

### The chainWebpack entry of the vue.config.js file
```
chainWebpack: config => {
    // Source folder
    config
    .entry("app")
    .clear()
    .add("./src/renderer/main.js")
    .end()
    // Copy preload script in dist (for production)
    config.entry('electron-preload')
    .add('./src/main/preload.js')
    .end()
  }
```

The pluginOptions entry of the vue.config.js file
```
pluginOptions: {
    electronBuilder: { 
      builderOptions: {
        appId: 'xxxxxxxxxxx',
        productName: 'xxxxxx',
      },
      // Same configuration in the webPreferences of the BrowserWindow
      nodeIntegration: false,
      contextIsolation: true,
      enableRemoteModule: false,
      sandbox: true,
      mainProcessFile: 'src/main/index.js',
      preload: 'src/main/preload.js',
    }
```

Does anyone has any idea of what I'm doing wrong? Or of what I should configure differently? 
Thank you very much!

**EDIT**
Webpack doesn't replace require by __webpack_require__ in the production preload file.

**Comment by oz-** [comment]
For those interested, [this post ](https://stackoverflow.com/a/65242779/1060921) solved this issue: putting nodeIntegration and contextIsolation outside of the webPreferences subobject did the trick. 

But, I realized ipcRenderer's listener ('on') wasn't exported to the shared object. 

So, I had to use invoke/handle (see [here](https://stackoverflow.com/a/62630044/1060921) and [here](https://github.com/electron/electron/issues/21437)) and change the way I was handling communication between the main and the renderer processes. 

At the end, my BrowserWindow configuration looks like this: 

```
        ...
        nodeIntegration: false,
        contextIsolation: true,
        enableRemoteModule: false,
        sandbox: true,

        webPreferences: {
          nodeIntegration: false,
          contextIsolation: true,
          enableRemoteModule: false,
          sandbox: true,
          preload: path.resolve(__dirname, 'preload.js')
        }
```
